BenadrylAutomation/src/main/java/com/jnj/pageobjects/ListingPage.java

```diff
@@ -0,0 +1,151 @@
+
+	@FindBy(css = "[data-value='61']")
+	private WebElement kidsProdQuickFilter;
+
+	@FindBy(xpath = "//div[@id='onetrust-close-btn-container']")
+	private WebElement closePrivacyBtn;
+
+	@FindBy(css = ".main-row .three-quarters")
+	private WebElement productsPanel;
+
+	@FindBy(id = "edit-sort-by")
+	private WebElement sortBySelector;
+
+	@FindBy(css = ".sfmc-careclub-lightbox-close")
+	private WebElement closeSignUpBtn;
+
+	private WebElement getBanner(String text) {
+		return driver.findElement(By.xpath("//div[@class='inner']/*[contains(.,'" + text + "')]"));
+	}
+
+	private WebElement getFilter(String text) {
+		return driver.findElement(By.xpath("//div[@class='filter-row-shortcuts']/a[contains(.,'" + text + "')]"));
+	}
+
+	/**
+	 * Function to click on close privacy pop-up button
+	 */
+	public void closePrivacyPopup() {
+		Action.explicitWaitForElementTobeclickable(closePrivacyBtn, 30);
+		boolean eleDisplayed = closePrivacyBtn.isDisplayed();
+		if (eleDisplayed) {
+			extentPassLog("Privacy pop-up displayed : ", true);
+			Action.performActionwithExtentInfoLog(closePrivacyBtn, "click", "Clicking on : Close button");
+		} else
+			extentFailLog("Privacy pop-up displayed : ", false);
+	}
+
+	/**
+	 * Function to click on close sign up pop-up button
+	 */
+	public void closeSignupPopup() {
+		Action.explicitWaitForElementTobeclickable(closeSignUpBtn, 30);
+		boolean eleDisplayed = closeSignUpBtn.isDisplayed();
+		if (eleDisplayed) {
+			extentPassLog("Privacy pop-up displayed : ", true);
+			Action.performActionwithExtentInfoLog(closeSignUpBtn, "click", "Clicking on : Close button");
+		} else
+			extentFailLog("Signup pop-up displayed : ", false);
+	}
+
+	/**
+	 * Visits a product listing page from the homepage.
+	 */
+	public void visitPLP() {
+		Action.performActionwithExtentInfoLog(prodHeaderMenu, "click", "Clicking on : " + prodHeaderMenu.getText());
+	}
+
+	/**
+	 * Visits a listing page from the homepage. Then, verifies the banner image in
+	 * the listing page.
+	 */
+	public void verifyBannerImage() {
+		this.visitPLP();
+		extentInfoLog("Image is displayed : ", Action.isDisplayed(driver, bannerImage));
+	}
+
+	/**
+	 * Visits a listing page from the homepage. Then, verifies the banner texts in
+	 * the listing page.
+	 */
+	public void verifyBannerTexts(String text) {
+		this.visitPLP();
+		WebElement actualElem = getBanner(text);
+		String actualText = actualElem.getText();
+		extentInfoLog("Banner heading is displayed : ", Action.isDisplayed(driver, actualElem));
+		Assert.assertEquals(actualText, text);
+		extentInfoLog("Banner text verified");
+	}
+
+	/**
+	 * Visits a listing page from the homepage. Then, verifies the filters in the
+	 * listing page.
+	 */
+	public void verifyQuickFilters(String text) {
+		this.visitPLP();
+		WebElement actualElem = getFilter(text);
+		String actualText = actualElem.getText();
+		extentInfoLog("Filter text is displayed : ", Action.isDisplayed(driver, actualElem));
```

Comment:
use proper assertions to check if element is displayed.
Assert.assertTrue(ele.isDsiplayed())

Reply:
Assertion added

Reply:
Verified